### PR TITLE
chore(client): Update client to Angular v6 + RxJS v6 (compat)

### PR DESCRIPTION
**Pull request by PascalPrecht**
This does not yet update Angular CLI to v6

### CODE REVIEW

File: client/src/app/auth/firebase-auth.service.ts [hunk 1]
```diff
@@ -2,9 +2,7 @@ import * as firebase from 'firebase';
 
 import { Injectable } from '@angular/core';
 
-import { Observable } from 'rxjs/Observable';
-import { fromPromise } from 'rxjs/observable/fromPromise';
-import { of } from 'rxjs/observable/of';
+import { Observable, from as fromPromise, of } from 'rxjs';
```
Comment (SamVerschueren) [inline client/src/app/auth/firebase-auth.service.ts]: Just wondering why we wouldn't go for just `from` and change the code?

File: client/package.json [hunk 2]
```diff
@@ -48,26 +48,27 @@
     "firebase": "^4.12.0",
     "js-yaml": "^3.8.4",
     "ng2-slim-loading-bar": "^4.0.0",
-    "ngx-clipboard": "^9.0.0",
-    "ngx-monaco": "0.6.0",
-    "rxjs": "^5.5.6",
+    "ngx-clipboard": "11.0.0",
+    "ngx-monaco": "0.8.0",
+    "rxjs": "6.1.0",
+    "rxjs-compat": "6.1.0",
```
Comment (SamVerschueren) [inline client/package.json]: Do we really need the compat library? Or is it for dependencies who rely on our `rxjs` version and still use the old syntax?

File: client/src/app/auth/firebase-auth.service.ts (hunk 1 above)
Comment (PascalPrecht) [reply client/src/app/auth/firebase-auth.service.ts]: This was done by `rxjs-tslint` so I just went with it.

File: client/package.json (hunk 2 above)
Comment (PascalPrecht) [reply client/package.json]: Exactly. Shared code is still on an older version, so is server/admin-cli/cli etc.

**Comment by PascalPrecht** [comment]
Thanks for the review @SamVerschueren!

**Comment by PascalPrecht** [comment]
I'll wait for #816 to land and rebase this on top of it.

**Pull request merged**